Upstream-status: Backport [https://crrev.com/c/1213155]

Signed-off-by: Maksim Sisov <msisov@igalia.com>
---
From 5917a8e6a1f8bb8679feecd5f0248aff097e9fdd Mon Sep 17 00:00:00 2001
From: Maksim Sisov <msisov@igalia.com>
Date: Fri, 14 Sep 2018 04:32:30 +0000
Subject: [PATCH] Fix 'constexpr' needed for in-class initialization of static
 data.

This fixes the following GCC error:

error: 'constexpr' needed for in-class initialization of static data

chrome/browser/browser/tab_manager_stats_collector.o
../../chrome/browser/resource_coordinator/tab_manager_stats_collector.cc:644:49:

member 'const base::TimeDelta
resource_coordinator::TabManagerStatsCollector::kLowFrequencySamplingInterval'
of non-integral type [-fpermissive]
 const base::TimeDelta
                                                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~

error: 'constexpr' needed for in-class initialization of static data
TabManagerStatsCollector: :kLowFrequencySamplingInterval;
Change-Id: I5ddcd2ef64c22e88a21d157f125c930d54a155ac
Reviewed-on: https://chromium-review.googlesource.com/1213155
Commit-Queue: Chris Hamilton <chrisha@chromium.org>
Reviewed-by: Chris Hamilton <chrisha@chromium.org>
Cr-Commit-Position: refs/heads/master@{#591288}
---
 chrome/browser/resource_coordinator/tab_manager_stats_collector.cc | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/chrome/browser/resource_coordinator/tab_manager_stats_collector.cc b/chrome/browser/resource_coordinator/tab_manager_stats_collector.cc
index 85bef76171fb..2fdb88b1b049 100644
--- a/chrome/browser/resource_coordinator/tab_manager_stats_collector.cc
+++ b/chrome/browser/resource_coordinator/tab_manager_stats_collector.cc
@@ -641,6 +641,7 @@ const char
         "TabManager.SessionOverlap.BackgroundTabOpening";
 
 // static
-const base::TimeDelta TabManagerStatsCollector::kLowFrequencySamplingInterval;
+constexpr base::TimeDelta
+    TabManagerStatsCollector::kLowFrequencySamplingInterval;
 
 }  // namespace resource_coordinator
-- 
2.11.0

